torchchat/utils/quantize.py

```diff
@@ -110,21 +119,30 @@ def quantize_model(
         if quantizer not in quantizer_class_dict:
             raise RuntimeError(f"unknown quantizer {quantizer} specified")
         else:
+            ao_quant = True
             # Use tensor subclass API for int4 weight only.
             if device == "cuda" and quantizer == "linear:int4":
                 quantize_(model, int4_weight_only(q_kwargs["groupsize"]))
+            elif quantizer == "linear:int8":
+                print("quantizer is linear int8")
+                quantize_(model, int8_weight_only())
```

Comment:
Why not integrate it into a QuantHandler class dispatched thru the handler dict at a single call site rather than build a chain of if statements?

Reply:
Hi @mikekgfb, we will refactor this part in the future after all quant APIs are moved to torchao I think

Reply:
torchAO already has a class-based API that is used for other quantizers?  Why do these differently, and then later refactor them?   Or why not do them all a consistent way now, and if you refactor later, do that?

Reply:
yeah, quantizer API is deprecated in favor of `quantize_`, that's why we are gradually refactoring the quantizer APIs to use `quantize_`, the reason we do it one by one is because there might be missing support/alignment on numerics etc. that we need to do during the migration